### PR TITLE
Setup Argparse + multiple scripts

### PR DESCRIPTION
**Pull request by 0xHericles**
### Features
This PR introduces two new features to tutorterminal
#### Argparse
Argparse + help text
```
usage: learn_script.py [-h] files [files ...]

Run Tutorterminal application.

positional arguments:
  files       the script files to run

optional arguments:
  -h, --help  show this help message and exit
```

#### Multiples scripts 
You can now run tutorials in sequence:
```
python3 learn_script.py example_git_init.txt example_git_commit.txt
```
And the _tutorterminal_ shows the progress and which tutorial you are running
```
Running script [1/2] - 'example_git_init.txt'
```

### CODE REVIEW

File: learn_script.py [hunk 1]
```diff
@@ -4,39 +4,36 @@
...
+import argparse
 
 try:
     from ansimarkup import parse, ansiprint
 except:
-    ansiprint = lambda x: print(re.sub('<.*?>', '', x))
-
-
-script = open(sys.argv[1]).readlines()
+    def ansiprint(x): return print(re.sub('<.*?>', '', x))
```
Comment (matheusgr) [inline learn_script.py]: avoid def call on one-liners, specially on import definitions.

```suggestion
    ansiprint = lambda x: print(re.sub('<.*?>', '', x))
```

File: learn_script.py [hunk 2]
```diff
@@ -4,39 +4,36 @@
...
 
 def prompt():
-    ansiprint("<bold><green>[In]</green></bold> <light-blue>" + os.path.abspath('.') + "</light-blue>$ ", end='')
+    ansiprint("<bold><green>[In]</green></bold> <light-blue>" +
+              os.path.abspath('.') + "</light-blue>$ ", end='')
 
 
 def query_user(content):
-    fancy_input = lambda : prompt() or input()
+    def fancy_input(): return prompt() or input()
```
Comment (matheusgr) [inline learn_script.py]: avoid function definition inside other functions and on one-liners.

```suggestion
    fancy_input = lambda : prompt() or input()
```

File: learn_script.py [hunk 3]
```diff
@@ -4,39 +4,36 @@
...
 try:
     from ansimarkup import parse, ansiprint
 except:
-    ansiprint = lambda x: print(re.sub('<.*?>', '', x))
-
-
-script = open(sys.argv[1]).readlines()
+    def ansiprint(x): return print(re.sub('<.*?>', '', x))
 
-help_txt = []
```
Comment (matheusgr) [inline learn_script.py:17]: did you test this remove? this seems to cause an error on line 107

File: learn_script.py [hunk 4]
```diff
@@ -61,61 +58,82 @@ def run_command(content, free=False, auto=False):
...
+        if command == 'run':
+            if not run_command(content):
+                exit(1)
+        if command == 'run_auto':
+            if not run_command(content, auto=True):
+                exit(1)
+        if command == '###':
+            processing = content
+
+if __name__ == "__main__":
```
Comment (matheusgr) [inline learn_script.py]: create a main function to call.

File: learn_script.py [hunk 5]
```diff
@@ -61,61 +58,82 @@ def run_command(content, free=False, auto=False):
...
+    parser = argparse.ArgumentParser(description="Run Tutorterminal application.")
+
+    parser.add_argument('files', metavar='files', type=str, nargs='+',
+                        help='the script files to run')
+
+    args = parser.parse_args()
+    num_scripts = len(args.files)
```
Comment (matheusgr) [inline learn_script.py]: validate files before calling:

```suggestion
    invalid_files = list(filter(lambda x: x[0], map(lambda x: (not os.path.isfile(x), x), args.files)))
    if invalid_files:
        for _, fname in invalid_files:
             print(fname + " not found")
    num_scripts = len(args.files)
```

**Review by matheusgr — CHANGES_REQUESTED** [review]
see requests. please test with examples scripts on root.

File: learn_script.py (hunk 1 above)
Comment (0xHericles) [reply learn_script.py]: This is black's fault, haha. I'll fix it.

File: learn_script.py (hunk 2 above)
Comment (0xHericles) [reply learn_script.py]: :+1:

File: learn_script.py (hunk 3 above)
Comment (0xHericles) [reply learn_script.py:17]: I tested it, and everything went well here :thinking:

File: learn_script.py (hunk 4 above)
Comment (0xHericles) [reply learn_script.py]: :+1:

File: learn_script.py (hunk 5 above)
Comment (0xHericles) [reply learn_script.py]: Nice!

**Comment by 0xHericles** [comment]
Ready, @matheusgr! :rocket:

**Review by matheusgr — APPROVED** [review]

**Pull request merged**